h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/BoosterParms.java

```diff
@@ -0,0 +1,56 @@
+package hex.tree.xgboost;
+
+import water.Iced;
+import water.util.IcedHashMapGeneric;
+
+import java.text.DecimalFormat;
+import java.text.NumberFormat;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Iced Wrapper around Booster parameter map. The main purpose is to avoid mistakes when using the parameter
+ * object directly: this class ensures that returned parameters will be localized.
+ */
+public class BoosterParms extends Iced<BoosterParms> {
+
+  private IcedHashMapGeneric.IcedHashMapStringObject _parms;
+
+  public static BoosterParms fromMap(Map<String, Object> map) {
+    BoosterParms bp = new BoosterParms();
+    bp._parms = new IcedHashMapGeneric.IcedHashMapStringObject();
+    bp._parms.putAll(map);
+    return bp;
+  }
+
+  /**
+   * @return localized Booster parameters
+   */
+  public Map<String, Object> get() {
+    return localizeDecimalParams(_parms);
+  }
+
+  /**
+   * Iterates over a set of parameters and applies locale-specific formatting
+   * to decimal ones (Floats and Doubles).
+   *
+   * @param params Parameters to localize
+   * @return Map with localized parameter values
+   */
+  private static Map<String, Object> localizeDecimalParams(final Map<String, Object> params) {
+    Map<String, Object> localized = new HashMap<>(params.size());
+    final NumberFormat localizedNumberFormatter = DecimalFormat.getNumberInstance();
+    for (String key : params.keySet()) {
```

Comment:
Iterating over entrySet should be faster than first obtaining set of keys and then call get() for each key separately (even if the lookup it is constant). Lots of other methods calls.

Reply:
Please keep in mind that the backing map is an instance of NonBlockingHashMap.

Javadoc of NBHM for the entrySet() method says this:

    <p><strong>Warning:</strong> the iterator associated with this Set
    requires the creation of {@link java.util.Map.Entry} objects with each
    iteration.  The {@link NonBlockingHashMap} does not normally create or
    using {@link java.util.Map.Entry} objects so they will be created soley
    to support this iteration.  Iterating using {@link #keySet} or {@link
    #values} will be more efficient.